Assets/Mirror/Core/NetworkConnection.cs

```diff
@@ -165,37 +197,51 @@ internal virtual void Send(ArraySegment<byte> segment, int channelId = Channels.
+                    // validate packet before handing the batch to the
+                    // transport. this guarantees that we always stay
+                    // within transport's max message size limit.
+                    // => just in case transport forgets to check it
+                    // => just in case mirror miscalulated it etc.
+                    ArraySegment<byte> segment = writer.ToArraySegment();
+                    if (ValidatePacketSize(segment, channelId))
+                    {
+                        // send to transport
+                        SendToTransport(segment, channelId);
+                        //UnityEngine.Debug.Log($"sending batch of {writer.Position} bytes for channel={kvp.Key} connId={connectionId}");
+
+                        // reset writer for each new batch
+                        writer.Position = 0;
+                    }
+                }
+            }
+        }
+
         // flush batched messages at the end of every Update.
         internal virtual void Update()
         {
+            if (reliableBatcher != null)
+            {
+                UpdateBatcher(Channels.Reliable, reliableBatcher);
+            }
+            
+            if (unreliableBatcher != null)
+            {
+                UpdateBatcher(Channels.Unreliable, unreliableBatcher);
+            }
+
             // go through batches for all channels
             // foreach ((int key, Batcher batcher) in batches) // Unity 2020 doesn't support deconstruct yet
             foreach (KeyValuePair<int, Batcher> kvp in batches)
             {
-                // make and send as many batches as necessary from the stored
-                // messages.
-                using (NetworkWriterPooled writer = NetworkWriterPool.Get())
-                {
-                    // make a batch with our local time (double precision)
-                    while (kvp.Value.GetBatch(writer))
-                    {
-                        // validate packet before handing the batch to the
-                        // transport. this guarantees that we always stay
-                        // within transport's max message size limit.
-                        // => just in case transport forgets to check it
-                        // => just in case mirror miscalulated it etc.
-                        ArraySegment<byte> segment = writer.ToArraySegment();
-                        if (ValidatePacketSize(segment, kvp.Key))
-                        {
-                            // send to transport
-                            SendToTransport(segment, kvp.Key);
-                            //UnityEngine.Debug.Log($"sending batch of {writer.Position} bytes for channel={kvp.Key} connId={connectionId}");
-
-                            // reset writer for each new batch
-                            writer.Position = 0;
-                        }
-                    }
-                }
+                UpdateBatcher(kvp.Key, kvp.Value);
```

Comment:
why have this loop when we already update both batches above the loop?
wouldn't it make sense to remove .batches completely?

Reply:
Need to remove any channels other than Reliable/Unreliable completely then?
Maybe save that for another time..